db/blob/db_blob_basic_test.cc

```diff
@@ -267,6 +267,51 @@ TEST_F(DBBlobBasicTest, GenerateIOTracing) {
 }
 #endif  // !ROCKSDB_LITE
 
+TEST_F(DBBlobBasicTest, BestEffortsRecovery_MissingNewestBlobFile) {
```

Comment:
Originally, I added the compaction because I would like to demonstrate that no new blob files are generated during this test even with compaction, but I think you are right too. It's not necessary.